src/Adapter/Container/DoctrineBuilderExtension.php

```diff
@@ -27,8 +27,9 @@
 namespace PrestaShop\PrestaShop\Adapter\Container;
```

Comment:
I wonder why this is in Adapter... shouldn't it be in the bundle?

Reply:
Because it is used by our `ContainerBuilder` which namespace is in Adapter

This class is not an extension as in a Symfony extension, it's a custom extension used by our `ContainerBuilder` only, which implements our `PrestaShop\PrestaShop\Adapter\Container\ContainerBuilderExtensionInterface` So it shouldn't be in the bundle as it's not related to Symfony framework and can't be used by it

An extra pre layer was required additionally to the `CompilerPassInterfaces` because some actions need to be done before the container is built and start processing the compiler passes So I introduced this extension notion to manage Doctrine, but from the recent modification from this PR I actually wonder if we could have used the Symfony extension system instead

Also having a centralized/common config parsed by the container builder and passed to extensions might have been nice (that's the Symfony principle), but it represents a more in depth refacto

Reply:
Oh I see.